transformer_engine/jax/csrc/utils.h

```diff
@@ -26,25 +27,49 @@ void PopulateRngStateAsync(void *rng_state_dst, const void *const seed, size_t q
                            size_t kv_max_seqlen, NVTE_Fused_Attn_Backend backend,
                            cudaStream_t stream);
 
-class cublasLtMetaManager {
+class WorkspaceManager {
  public:
-    static cublasLtMetaManager &Instance() {
-        static thread_local cublasLtMetaManager instance;
+    static WorkspaceManager &Instance() {
+        static thread_local WorkspaceManager instance;
         return instance;
     }
 
-    cublasLtMetaManager() {}
-    ~cublasLtMetaManager() { Clear_(); }
+    WorkspaceManager() {}
+    ~WorkspaceManager() { Clear_(); }
 
     void *GetWorkspace(size_t size = 4194304) {
         ReallocateIfNeed_(size);
         return workspace_;
     }
 
+    std::vector<void *> GetWorkspace(std::vector<size_t> sizes) {
```

Comment:
Since the number of asked workspace sizes are fixes, I would prefer to use fixed sizes container like `std::initializer_list` or `std::array` to avoid the dynamic heap allocations. Here is an example

```
    template<typename... Args>
    inline auto GetWorkspace(Args... args) {
        auto asks = std::array<size_t, sizeof...(Args)>{args...};
        std::array<size_t, sizeof...(Args) + 1> offsets = {0};
        std::array<void*, sizeof...(Args)> workspaces = {nullptr};

        std::transform_inclusive_scan(
            asks.cbegin(), asks.cend(), offsets.begin() + 1,
            std::plus<size_t>{}, [=](auto x) {return PadSize_(x);}, 0);

        auto* workspace = GetWorkspace(offsets.back());

        std::transform(
            offsets.cbegin(), offsets.cend() - 1, workspaces.begin(),
            [workspace](auto x) {return static_cast<char*>(workspace) + x;});

        return workspaces;
    }
```
```